lib/ui/painting/image_descriptor.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/image_descriptor.h"
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+#include "flutter/lib/ui/painting/codec.h"
+#include "flutter/lib/ui/painting/image_decoder.h"
+#include "flutter/lib/ui/painting/multi_frame_codec.h"
+#include "flutter/lib/ui/painting/single_frame_codec.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "third_party/tonic/dart_binding_macros.h"
+#include "third_party/tonic/logging/dart_invoke.h"
+
+namespace flutter {
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, ImageDescriptor);
+
+#define FOR_EACH_BINDING(V)            \
+  V(ImageDescriptor, initRaw)          \
+  V(ImageDescriptor, instantiateCodec) \
+  V(ImageDescriptor, width)            \
+  V(ImageDescriptor, height)           \
+  V(ImageDescriptor, bytesPerPixel)
+
+FOR_EACH_BINDING(DART_NATIVE_CALLBACK)
+
+void ImageDescriptor::RegisterNatives(tonic::DartLibraryNatives* natives) {
+  natives->Register(
+      {{"ImageDescriptor_initEncoded", ImageDescriptor::initEncoded, 3, true},
+       FOR_EACH_BINDING(DART_REGISTER_NATIVE)});
+}
+
+void ImageDescriptor::initEncoded(Dart_NativeArguments args) {
```

Comment:
This used to be done asynchronously earlier but is now is potentially in a frame workload. So it is a minor regression. I'd add a trace here at the very least. Better still, put the codec generation on the concurrent worker pool. See my comment about using the callback to return the result. That should allow you to invoke the result back on the UI task runner when the worker is done with the same.

Reply:
Ahh we should definitely make this be done on the worker again, I'll fix that. It's already using a callback to let Dart side know when things are done though.